client/src/redux/files/fileSlice.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  createSlice,
+  PayloadAction,
+  createAsyncThunk,
+  ActionReducerMapBuilder
+} from '@reduxjs/toolkit'
+
+import fileService from './fileService'
+import { FileStateType } from './fileType'
+import { catchError } from '~/utils/handleAxiosError'
+import formatFiles from '~/utils/formatFiles'
+
+const initialState: FileStateType = {
+  files: [],
+  isError: false,
+  isLoading: false,
+  isSuccess: false,
+  error: {
+    status: 0,
+    content: null
+  }
+}
+
+export const getProjectFiles = createAsyncThunk(
+  'file/getProjectFiles',
+  async (projectId: number, thunkAPI) => {
+    try {
+      const response = await fileService.getProjectFiles(projectId)
+      const files = formatFiles(response)
+      return files
+    } catch (error) {
+      return thunkAPI.rejectWithValue(catchError(error))
+    }
+  }
+)
+
+export const getProjectFile = createAsyncThunk(
+  'file/getProjectFile',
+  async (payload: { projectId: number; fileId: string }, thunkAPI) => {
+    try {
+      const response = await fileService.getProjectFile(payload.projectId, payload.fileId)
+      return response
+    } catch (error) {
+      return thunkAPI.rejectWithValue(catchError(error))
+    }
+  }
+)
+
+export const addProjectFile = createAsyncThunk(
+  'file/addProjectFile',
+  async (payload: { projectId: number; file: any }, thunkAPI) => {
+    try {
+      const response = await fileService.addProjectFile(payload.projectId, payload.file)
+      const files = await fileService.getProjectFiles(payload.projectId)
+      const formattedFiles = formatFiles(files)
+      if (response === 'File uploaded successfully') {
+        return {
+          message: response,
+          files: formattedFiles
+        }
+      }
+      return thunkAPI.rejectWithValue({ status: 422, content: response, files: formattedFiles })
+    } catch (error) {
+      return thunkAPI.rejectWithValue(catchError(error))
+    }
+  }
+)
+
+export const renameProjectFile = createAsyncThunk(
+  'file/renameProjectFile',
+  async (payload: { projectId: number; fileId: string; name: string }, thunkAPI) => {
+    try {
+      const response = await fileService.renameProjectFile(
+        payload.projectId,
+        payload.fileId,
+        payload.name
+      )
+      if (response === 'File updated successfully') {
+        const files = await fileService.getProjectFiles(payload.projectId)
+        const formattedFiles = formatFiles(files)
+        return formattedFiles
+      }
+    } catch (error) {
```

Comment:
Do you mean `response.data.message === 'File updated successfully'`?

Reply:
From our previous review of implemented axios methods yesterday, `renameProjectFile` returns a `Promise<string>`

```
const renameProjectFile = async (
  projectId: number,
  fileId: string,
  name: string
): Promise<string>
```